thirdeye/thirdeye-frontend/app/pods/self-serve/create-alert/controller.js

```diff
@@ -4,29 +4,647 @@
+      granularities: fetch(`/data/agg/granularity/metric/${metricId}`).then(res => res.json()),
+      filters: fetch(`/data/autocomplete/filters/metric/${metricId}`).then(res => res.json()),
+      selectedMetricDimensions: fetch(`/data/autocomplete/dimensions/metric/${metricId}`).then(res =>res.json()),
+    };
+
+    return Ember.RSVP.hash(promiseHash);
+  },
+
   /**
-   * Placeholder for patterns of interest options
+   * Fetches the time series data required to display the anomaly detection graph for the current metric.
+   * @method fetchAnomalyGraphData
+   * @param {Object} config - key metric properties to graph
+   * @return {Promise} Returns time-series data for the metric
    */
-  patternsOfInterest: ['Pattern One', 'Pattern Two'],
-  selectedPattern: 'Pattern One',
+  fetchAnomalyGraphData(config) {
+    this.set('loading', true);
+
+    const {
+      id,
+      dimension,
+      currentStart,
+      currentEnd,
+      baselineStart,
+      baselineEnd,
+      granularity,
+      filters
+    } = config;
+
+    const url = `/timeseries/compare/${id}/${currentStart}/${currentEnd}/${baselineStart}/${baselineEnd}?dimension=${dimension}&granularity=${granularity}&filters=${filters}`;
+
+    return fetch(url).then(res => res.json());
+  },
+
   /**
-   * Placeholder for dimensions options
+   * Send a POST request to the entity endpoint to create a new record.
+   * @method saveThirdEyeEntity
+   * @param {Object} alertData - The record being saved (in this case, a new alert config group)
+   * @param {String} entityType - The type of entity being saved
+   * @return {Promise}
    */
-  dimensionExploration: ['First', 'Second', 'Third'],
-  selectedDimension: 'Select One',
+  saveThirdEyeEntity(alertData, entityType) {
+    const postProps = {
+      method: 'post',
+      body: JSON.stringify(alertData),
+      headers: { 'content-type': 'Application/Json' }
+    };
+    const url = '/thirdeye/entity?entityType=' + entityType;
+    return fetch(url, postProps);
+  },
+
   /**
-   * Placeholder for alert groups options
+   * Send a POST request to the new function create endpoint.
+   * @method saveThirdEyeFunction
+   * @param {Object} functionData - The new function to save
+   * @return {Promise}
    */
-  allAlertsConfigGroups: Ember.computed.reads('model.allAlertsConfigGroups'),
-  selectedGroup: 'Select an alert group, or create new',
+  saveThirdEyeFunction(functionData) {
+    const postProps = {
+      method: 'post',
+      headers: { 'content-type': 'Application/Json' }
+    };
+    const url = '/dashboard/anomaly-function/create?' + this.toQueryString(functionData);
+    return fetch(url, postProps).then(res => res.json());
+  },
+
+  /**
+   * Loads time-series data into the anomaly-graph component
+   * @method triggerGraphFromMetric
+   * @param {Number} metricId - Id of selected metric to graph
+   * @return {undefined}
+   */
+  triggerGraphFromMetric(metricId) {
```

Comment:
This could also potentially be a computed property watching the fields that you care about